emission/net/ext_service/habitica/auto_tasks/level_challenge.py

```diff
@@ -0,0 +1,210 @@
+		timestamp_now = arrow.now().timestamp
+
+		#Get metrics
+		summary_ts = earmt.group_by_timestamp(userid, timestamp_from_db, timestamp_now, None, earmts.get_count)
+	    logging.debug("Metrics response: %s" % summary_ts)
+
+	    for item in summary_ts["result"]
+	    	try:
+	            bike_trips += item.BICYCLING
+	            logging.debug("bike_trips += %s" % item.BICYCLING)
+	        except AttributeError:
+	            logging.debug("no bike")
+
+	    logging.debug("Finished with bike_trips == %s" % bike_trips)
+
+	    method_uri_level_challenge = "/api/v3/tasks/"+ task.task_id + "/score/up"
+	    
+	    #reward user by scoring + habits
+	    #1 trip bike is +1
+	    for item in range(bike_trips):
+	    	res = proxy.habiticaProxy(user_id, 'POST', method_uri_level_challenge, None)
+	    	logging.debug("Request to score bike points %s" % res)
+
+	    #update the bike count ts
+	    new_state = {'last_timestamp': summary_ts["last_ts_processed"]}
+
+	    logging.debug("Returning %s" % new_state)
+	    return new_state
+
+	# Level 3 - 100 km on bike (1 score up for each km)
+	if len(task.habitica_task.history) >= trips_to_cover + level_1_repetition and len(task.habitica_task.history) < distance_to_cover + trips_to_cover + level_1_repetition:
+		
+		bike_scale = task["args"]["bike_distance_scale"]
+
+		if len(task.habitica_task.history) == 11:
+	        leftover_bike = 0 
+	    else:
+	        leftover_bike = curr_state['bike_leftover']
+
+	    timestamp_from_db = curr_state['last_timestamp']
+	    timestamp_now = arrow.now().timestamp
+
+	    #Get metrics
+	    summary_ts = earmt.group_by_timestamp(user_id, timestamp_from_db, timestamp_now, None, earmts.get_distance)
+	    logging.debug("Metrics response: %s" % summary_ts)
+	    
+	    bike_distance = leftover_bike
+	    for item in summary_ts["result"]:
+	        try:
+	            bike_distance += item.BICYCLING
+	            logging.debug("bike_distance += %s" % item.BICYCLING)
+	        except AttributeError:
+	            logging.debug("no bike")
+
+
+	    logging.debug("Finished with bike_distance == %s" % bike_distance)
+
+	    method_uri_level_challenge = "/api/v3/tasks/"+ task.task_id + "/score/up"
+	    
+	    #reward user by scoring + habits
+	    #1 km bike is +1
+	    bike_pts = int(bike_distance//bike_scale)
+
+	    for i in range(bike_pts):
+	    	res = proxy.habiticaProxy(user_id, 'POST', method_uri_level_challenge, None)
+	    	logging.debug("Request to score bike points %s" % res)
+
+	    #update the bike distance
+	    new_state = {'last_timestamp': summary_ts["last_ts_processed"],
+	    			'bike_leftover': bike_distance%bike_scale}
+
+	    logging.debug("Returning %s" % new_state)
+	    return new_state
+
+	# Level 4 - 
+	if len(task.habitica_task.history) >= distance_to_cover + trips_to_cover + level_1_repetition: ######and complete == False:
+		
+		if len(task.habitica_task.history) == 111 :
+			done_for_day = False
+			ts_from_level_3 = curr_state["last_timestamp"]
```

Comment:
I'm not convince this is right, but I'm going to think about it tonight. In particular, I think that this assumes that the pipeline is running continuously. What if it does down for a bit? What if I want to re-run after the gap of a few days? Would be nice to make it easier to support that.